pkgs/build-support/rust/rustc-wrapper/rustc-wrapper.sh

```diff
@@ -1,6 +1,14 @@
 #!@shell@
 
-extraBefore=(@sysroot@)
+defaultSysroot=(@sysroot@)
+
+for arg; do
+    if [ "$arg" = --sysroot ]; then
+        defaultSysroot=()
```

Comment:
```suggestion
    if [[ "$arg" = --sysroot ]]; then
        defaultSysroot=()
        break
```

absolutely nitpicking here btw

Reply:
I'd rather not break, because we'll probably want to add other flags here in the future.  Given that, it should probably be a `case` rather than an `if` though.  I'll change that.